### PR TITLE
Update Report_ActionItem

### PR DESCRIPTION
**Pull request by Luke9389**
### Details

This PR adds the pending and error states to Report_ActionItem and uses them to set the opacity to 50%. 

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/211551

### CODE REVIEW

**Comment by Luke9389** [comment]
I'm putting this on HOLD while I verify what needs to come before it. I think there may need to be some API PRs that happen first (not 100% sure yet though).

**Comment by Luke9389** [comment]
OK, I verified that this one is safe to merge without the other changes happening. The unpassed props evaluate to false because of the `||` casting them to booleans. 👍

File: src/pages/home/report/reportActionPropTypes.js [hunk 1]
```diff
@@ -23,4 +23,11 @@ export default {
         // The ID of the iou transaction
         IOUTransactionID: PropTypes.string,
     }),
+
+    // Used to indicate the pending state.
+    // Pending means the addComment request is waiting to be sent to the server.
```
Comment (Justicea83) [inline src/pages/home/report/reportActionPropTypes.js:28]: This comment isn't very clear

**Review by Justicea83 — CHANGES_REQUESTED** [review]

File: src/pages/home/report/reportActionPropTypes.js (hunk 1 above)
Comment (Luke9389) [reply src/pages/home/report/reportActionPropTypes.js]: Ok, I'll try rewording it. 👍Thx

File: src/styles/StyleUtils.js [hunk 2]
```diff
@@ -398,6 +399,9 @@ function getReportActionItemStyle(isHovered = false) {
 
             // Warning: Setting this to a non-transparent color will cause unread indicator to break on Android
             : colors.transparent,
+        opacity: isPending
```
[the remark this answers is not in the document]
Comment (Luke9389) [reply src/styles/StyleUtils.js]: good point :)

**Comment by Luke9389** [comment]
Updated

**Review by roryabraham — APPROVED** [review]

**Comment by roryabraham** [comment]
All yours @Justicea83

**Review by Justicea83 — APPROVED** [review]
Code LGTM!

**Pull request merged**